packages/dev/core/src/Events/pointerEvents.ts

```diff
@@ -9,35 +9,35 @@ import type { Ray } from "../Culling/ray";
 /**
  * Gather the list of pointer event types as constants.
  */
-export class PointerEventTypes {
+export enum PointerEventTypes {

```

Comment:
Can you make it an object instead as enum takes more space when transpiled down (which is essentially why they were not used) ?

[Playground Link](https://www.typescriptlang.org/play?#code/KYDwDg9gTgLgBMAdgVwLZwAoQJaJsKAUQDckYAVATzGAGc4BvAKDlbgHoAqTltuTuOQAWwOJFz4oAEwgB3RAlJ442egDNsUYFLiyRCgIZiceAnABGwAMYRUdOAasxspAHRwAYtDioIyWsAANCrwqnAaWjp6SHAwInBSwMTYVqIwUAaItNjOEFnhULZwiBAWyDAwefSJYFq0ATqVDvAANsAGtPB5oubllQo1dQ3uXlCxflZCwTkq6prauvpiQpTZVgYtcDZ4jqH0qAaJujlCsfFS2ADmOdgAXgQj3jSI03vh81FLcaKdlC3+PgMAGt7GAVmsNls8jBdsc4mdRBdrs57lBXLw2Jx2BjMAB5ACSADlyIQAEoAEVxAHVCXAALxwAAMIEZAEZAjiuDw+PxBPFxKYoMgwIoyLN3pFFjEjALJOKSnAWnlLmZdi5gOieVicRgCcSyQBVDD0pksgBMHL4XJxvOEollBF8pFFyjCEQW0UMxgkZkmmRV9Bs0AuiAM+Fomr42r4uqJJNJAFlcQA1Qgm5mMgAslrY1q1fPtJkk0WAmySYrdHylXt8-lEJc2Qg6FmAMUKMPwUkjmOxMb18apAAlCIQADLplkADhzrDzUYL3sFYBSQJdb3dn2lPjop28tFqOVETfolhiy6sIJ05koCMXkm7rGjbFj+tJGHxAGEANIT1mMmccNwNoCHad4EDCIrlq6cySp6Di3hA5gAFbWPAx4tjElTIJMCyZDoWhtB0HonH48BSBklwPvwvbPv2ZLkAAgsaDLMma-6ckB+agQ60h+OYbQQWu4obtW8HfHAiEoU4cDoaeChYThOh4XABHtA0sSyCk9YkeUCQUVRT6sC+8aUgaABCo6EIxzGmpm7EAL5AA)

which one of the reason enum where not used historically. Agree it does not offer the same typings protection which is why the new preserveConstEnum  is being evaluated in another PR to workaround the size expansion issue of enums.

Maybe something like this [Playground Link](https://www.typescriptlang.org/play?#code/C4TwDgpgBACg9gSwHbAgJwKIDcIoCrjQC8UA3gFBRQD0AVLZVVbVHgBbRiIroAmcAdyRQIOFFAQBnKADMEaCLygCOwgIZQuyVGigAjCAGM4AWwjS1h4AhwA6KADE4uk3ACukiABoJwCdLkFJRVcKGAOKF5RBENoYDQ1JEkEazgk2TRTKCQ4fTdgYDTpKLAFSU8lQqg1PwAbCDVJPzToPXzC4RKyivsnXUK3QzYfFP9ZeUVlVU02EGTDNVqoYxRLPykoEzUo5RS2MIjeBABzFIQAL3Re501cEfWAieDp8OgmkFqPTbUAa3MZuYxRbLNLANa7cIHaBHU7WS5oWyMJi0ahIqAKbZpWogWAAeQAkgA5PAYABKABFcQB1QkALmybhMBjQAG40XQGExmKwIloeGg3GARGIHuMglNQho+ToxjkoLU0sd0NUrDYIIiuVAUWiMfwkNi8USSaSAKowelIRnMtlcjlorU8zjcHSuHDC3CiwKTELqTTO5VDRJK6TGZxHJA1cwarnarm6rE4mAE4lkgCyuIAahgLVb0DamHbNSx2E7tOgQhAlqIPWMvc9JZt3J4JZWoGxGvoIKFMmDULxo8jUXGGnqDUmjWSqQAJDAYAAyOaZefZ9Htxd5-rQYBiP3d4g2dYlvrMkn2N0kpRS0Hb0gMoW3hj+Sj0ONefrLCLXQ6Y8f1ieTxowPiADCADSi7WiunIxo6778mCQrVvujzij61RQlAcB6AAVkYfg3p2oQDEMkyJEoCj1I03p7O4fi8AkxwDsw35UL+Y4AWSeAAILmgyS6slBa6wdKfDuHo9QIXunpPEe6FvlhuFWG2HZ3sIxEcEoZHopWDQVGEAgxNABnhLRkQMUxWosdpmJ-oaKYUriJoAEJzhg3G8Za-FsgAvmy5AQAAHlwaB+CsTSwJu2AegQkCSPS8AflF+CEFAJAUFy472ZSNL0gADAFuUAIxeGimXGmaeUFQATCVGUcaS6ZZpVuUACy1UwZWTjO87NQAHO1VCdaSQFgZVhW5QNdnGu5lVVRNpX1ZSzmuTNUD5S183eeQ5AJfySXADFUZDUtLluTxqVQONLJAA) ?

Reply:
If you don't want to use enumerations and want to ensure that members are read-only, you can use the namespace + union type scheme, as follows：

[TypeScript Playground](https://www.typescriptlang.org/zh/play?target=99&module=1#code/FAOwhgtgpgzgDmAxlABABQPYEsQBcoBOAKgJ5yoDewKNKUAHnBgbiohiDKwCIYDuIFAF4UABgDc1WgyYs2HLigCqcYSgCMkgL7BguMqkw58xA2tEoAPhsnB2nVmABc6bHkKlyNEUfenyAHS8Arq+Jp5QQfyCIurAQA)

Reply:
This would prevent the use of Flags like values: [Playground Link](https://www.typescriptlang.org/zh/play?target=99&module=1&ssl=9&ssc=6&pln=9&pc=17#code/FAOwhgtgpgzgDmAxlABABQPYEsQBcoBOAKgJ5yoDewKNKUAHnBgbiohiDKwCIYDuIFAF4UABgDc1WgyYs2HLigCqcYSgCMk2nUbNW7TqxUAmNca3Tdcg4pUBmNQBZJAX2DBcZVJhz5iXtVEUAB8NEJRTUOd3YBtWMAAudGw8QlJyGhEfVP9yADpeAXDsv3SoPJVilNKvCrhzdxK02sLBEXVgIA) which is why the values are defined as 0x02 0x04 ...

Reply:
For the user who uses the enumeration as a bitmask, it should be his responsibility to declare the mask variable of number, not the enumeration type to be responsible for the compatibility of his mask variable type;

Reply:
I would honestly prefer to wait for https://github.com/BabylonJS/Babylon.js/pull/14076 to be done before getting further only for PointerEventTypes.

Reply:
agree

Reply:
I have an idea on how to solve this, so eventually moving to const enums might be a viable option. This will, however, have to wait a bit.